docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,262 @@
+---
+title: Keyboard handling
+description: A guide for handling common keyboard interactions.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+
+Keyboard handling is crucial for creating an excellent user experience in your Expo app. This guide covers common keyboard interactions and how to manage them effectively.
+
+React Native provides several keyboard-related APIs, with [`Keyboard`](https://reactnative.dev/docs/keyboard) and [`KeyboardAvoidingView`](https://reactnative.dev/docs/keyboardavoidingview) being the most frequently used. For more complex or custom keyboard interactions, you might consider using [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller), which is a library that offers advanced keyboard handling capabilities.
+
+## The basics
+
+### Keyboard events
+
+The `Keyboard` module from React Native allows you to listen for native events, react to them, and make changes to the keyboard, such as dismissing it.
+
+To listen for keyboard events, use the `Keyboard.addListener` method. This method accepts an event name and a callback function as arguments. The callback function is called with the event data when the keyboard is shown or hidden.
+
+```tsx HomeScreen.tsx
+import { useEffect } from 'react';
+import { Keyboard } from 'react-native';
+
+export default function HomeScreen() {
+  useEffect(() => {
+    const showSubscription = Keyboard.addListener('keyboardDidShow', handleKeyboardShow);
+
+    return () => {
+      showSubscription.remove();
+    };
+  }, []);
+
+  const handleKeyboardShow = event => {
+    // Handle keyboard show event here
+  };
+}
+```
+
+To dismiss the keyboard, use the `dismiss` method.
+
+```tsx DismissKeyboardButton.tsx
+import { Keyboard, Button } from 'react-native';
+
+export default function DismissKeyboardButton() {
+  return <Button title="Dismiss keyboard" onPress={Keyboard.dismiss} />;
+}
+```
+
+### Keyboard avoiding view
+
+The `KeyboardAvoidingView` is a component that automatically adjusts its height, position, or bottom padding based on the keyboard height to remain visible while the keyboard is displayed.
+
+```tsx HomeScreen.tsx
+import { KeyboardAvoidingView, TextInput } from 'react-native';
+
+export default function HomeScreen() {
+  return (
+    <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : undefined} style={{ flex: 1 }}>
```

Comment:
We should clarify what `undefined` means here ... I forget if it means that the keyboard avoiding view has no effect (which is often what you want with `adjustResize`) or it defaults to something. If what you want here changes when you adjust to `pan`, we should clarify that.

Broader goal here: we can improve a bit on the default RN `KeyboardAvoidingView` documentation to clarify what are typically good values to use for `behavior`. I would say this would be whatever properly keeps a single `TextInput` visible within a scrollable view on both iOS and Android with the default `adjustResize` soft input value is a good baseline.